application/src/test/java/io/skai/template/services/AdGroupServiceIntegrationTest.java

```
@@ -0,0 +1,231 @@
+    private static final Status AD_GROUP_PAUSED = Status.PAUSED;
+
+    @InjectMocks
+    private AdGroupServiceImpl adGroupService;
+
+    @Mock
+    private CampaignDao campaignDao;
+    @Mock
+    private AdGroupDao adGroupDao;
+
+    @Captor
+    private ArgumentCaptor<AdGroup> adGroupArgumentCaptor;
+    @Captor
+    private ArgumentCaptor<Long> adGroupIdArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<FieldValidationException> fieldValidationExceptionArgumentCaptor;
+
+    @Test
+    public void verifyWhenAdGroupCreated() {
+        final Campaign campaign = Campaign.builder()
+                .name(CAMPAIGN_NAME)
+                .ksName(CAMPAIGN_KS_NAME)
+                .status(CAMPAIGN_ACTIVE)
+                .build();
+
+        final AdGroup adGroup = AdGroup.builder()
+                .campaignId(CAMPAIGN_ID)
+                .name(AD_GROUP_NAME)
+                .status(AD_GROUP_ACTIVE)
+                .build();
+
+        when(campaignDao.findById(eq(CAMPAIGN_ID))).thenReturn(Optional.ofNullable(campaign));
+        when(adGroupDao.create(adGroup)).thenReturn(AD_GROUP_ID);
+
+        final long adGroupId = adGroupService.create(adGroup);
+
+        verify(adGroupDao).create(adGroupArgumentCaptor.capture());
+
+        final AdGroup adGroupCaptorValue = adGroupArgumentCaptor.getValue();
+
+        assertThat(adGroupId, is(AD_GROUP_ID));
+        assertThat(adGroupCaptorValue.getName(), is(AD_GROUP_NAME));
+        assertThat(adGroupCaptorValue.getStatus(), is(AD_GROUP_ACTIVE));
+        assertThat(adGroupCaptorValue.getCampaignId(), is(CAMPAIGN_ID));
+    }
+
+    @Test
+    public void verifyWhenAdGroupNotCreatedWithWrongCampaignId() {
+        final AdGroup adGroup = AdGroup.builder()
+                .campaignId(CAMPAIGN_WRONG_ID)
+                .name(AD_GROUP_NAME)
+                .status(AD_GROUP_ACTIVE)
+                .build();
+
+        when(campaignDao.findById(CAMPAIGN_WRONG_ID)).thenThrow(
```

you can return empty campaign and your code should throw this error, no?